Replace default editor font to NotoSansUI

volzhs opened the pull request:
Fix #10076, #5647

![image](https://user-images.githubusercontent.com/8281454/29751651-36fc2250-8b8b-11e7-8495-1927549df4d9.png)


**Look carefully at the name of a project.**

![image](https://user-images.githubusercontent.com/8281454/29751680-801ae750-8b8b-11e7-8d26-ec3083aeb221.png)
Latin letter's baseline is not same with others.

![image](https://user-images.githubusercontent.com/8281454/29751691-95bb787c-8b8b-11e7-8893-1442c110140b.png)
It's corrected with NotoSansUI-Regular.ttf instead of DroidSans.
And looks better.

without this change https://github.com/godotengine/godot/compare/2.1...volzhs:editor-font?expand=1#diff-0f2d5f8b708eeeac121058c71f5c1428R88
![image](https://user-images.githubusercontent.com/8281454/29751695-a24891ec-8b8b-11e7-9543-39f9218e8425.png)
window title label position is wrong.

volzhs committed:
Replace default editor font to NotoSansUI

Calinou commented:
This looks quite nicer, I like it.

Noshyaar commented:
I think the master branch needs this to actually fix #10076.

[COPYRIGHT.txt](https://github.com/godotengine/godot/blob/2.1/COPYRIGHT.txt) also needs updating?

volzhs commented:
![image](https://user-images.githubusercontent.com/8281454/29765527-7cb5759a-8c16-11e7-812d-f6a3fb8ec5a0.png)

Note that all UI contains text will be a little bit bigger.
upper is an old one and below is with noto font.

I'm not sure it's worth with 2.1.x.
but it would be better to handle proper margin after this PR merged on master branch.

volzhs commented:
I tried to update COPYRIGHT.txt but don't know what should be written about _Copyright_
```
Files: ./thirdparty/fonts/NotoSansUI_Regular.ttf
Comment: Noto Sans UI Regular font
Copyright: 2012-2017, Google Internationalization Team
License: OFL-1.1
```

would this be fine? based on https://github.com/googlei18n/noto-fonts information.

akien-mga commented:
Yes, that looks perfect for the copyright. Please also update https://github.com/godotengine/godot/blob/2.1/thirdparty/README.md#fonts with the relevant details (upstream URL, version/date/commit, etc.)

BTW, is DroidSans still used after this PR, or should it be removed?

volzhs commented:
@akien-mga 
I will update README.md too.
I am not sure about removing DroidSans.ttf. may I remove it?

akien-mga commented:
Well if nothing uses DroidSans.ttf, let's remove it yes.

groud commented:
Have you considered using blazed instead ? 
![2017-08-29-102719_540x73_scrot](https://user-images.githubusercontent.com/6093119/29811705-ae4d27d6-8ca4-11e7-986c-2991295eb9bf.png)

Just kidding ^^, this change is cool. I figured out the wrong text alignment while working on the 2d editor, this should fix it. Good work ! :)

volzhs committed:
Update source code pro font to 2.030

volzhs commented:
- Updated `thirdparty/README.md` for Noto Sans
- Removed unused `DroidSans.ttf` file
- Updated Source code pro font to 2.030 and `thirdparty/README.md` for source code pro

akien-mga commented on thirdparty/README.md:
```diff
@@ -17,9 +17,23 @@ TODO.
+- Version: 1.06
+- License: OFL-1.1
+
+#### What’s the difference between the UI and non-UI versions?
```
So do we use the UI or non-UI version? :)

Why is this comment relevant for us?

volzhs replied:
we use UI version which has tighter vertical space.
I made this comment for somebody not to confuse which should be used between UI and Non-UI version in the future.

akien-mga commented:
What about the Arabic, Thai, etc. DroidSans variants? Do we still use those, and if yes should they be replaced by Noto too? Or is DroidSans better for those?

volzhs commented:
yes, we can update Arabic, Hebrew, Thai also.
not Fallback.ttf or Japanese.ttf because Noto fonts for this have huge file size.
I will test those 3 ttf with Noto fonts.

volzhs commented:
![image](https://user-images.githubusercontent.com/8281454/29816525-571bc81e-8cef-11e7-8b4c-a85a374c4759.png)
Test result with Noto fonts for Latin, Thai, Hebrew, Arabic.

it's wrong rendering for Hebrew, Arabic though.
here is original string.
> "Hello สวัสดี שלום مرحبا 안녕 こんにちは 你好"

what do @akien-mga and others think?

Noshyaar commented:
It's wrong because RTL isn't yet supported in Godot?

volzhs commented:
@Noshyaar that's correct.
And additional test with Noto CJK(16.4MB) instead of DroidSansFallback(3.6MB) and DroidSansJapanese(1.2MB).

![image](https://user-images.githubusercontent.com/8281454/29817704-05d93a40-8cf4-11e7-8e89-66145c39bb98.png)

HummusSamurai commented:
@volzhs The arabic script in Noto is much better than DroidSans. <3

volzhs commented:
I compiled release_debug tool with All noto fonts.
it's 40.6 MB with DroidSans.
and 52.6 MB with NotoFonts (all editor fonts are replaced with Noto)

akien-mga commented:
> I compiled release_debug tool with All noto fonts on Ubuntu.
it's 40.6 MB with DroidSans.
and 52.6 MB with NotoFonts (all editor fonts are replaced with Noto)

That's not worth it IMO. How is it like with only Noto for the main font, and with Noto for everything but CJK+Fallback?

volzhs commented:
@akien-mga it's 41.0 MB in case of all noto but not CJK+fallback.

akien-mga commented:
Good, let's go with that then.

volzhs commented:
@akien-mga

Fonts | Tools file size
--- | ---
All droid sans | 40.6 MB
noto for main | 40.7 MB
noto for all but CJK | 41.0 MB
all noto | 52.6 MB

volzhs committed:
Use Noto fonts for Arabic, Hebrew and Thai

volzhs commented:
Arabic, Hebrew, and Thai also use Noto fonts now.

Noshyaar commented:
I'll use this chance to leave my opinion about Thai font here. Although I don't think Thais make a significant number in Godot user base.

Thai DroidSans and Noto probably are the same font. Both are borderline visually appealing, a bit better than Tahoma, Leelawadee and Dokchampa font distributed with Windows.

Examples of, in my opinion, good looking Thai font for UI:
- Waree and Garuda (part of [fonts-tlwg](https://linux.thai.net/projects/fonts-tlwg) ([Github](https://github.com/tlwg/fonts-tlwg)))
- [CS ChatThai UI](http://www.f0nt.com/release/cs-chatthai/)
- [CS Prakas](http://www.f0nt.com/release/cs-prakas/)

(FYI, we have "[national fonts](http://www.f0nt.com/release/13-free-fonts-from-sipa/)" ([Wikipedia](https://en.wikipedia.org/wiki/National_fonts)) that are used in documents. Most are okay to use for official papers and printing, but all of them look horrible if used as-is in UI, especially with small size.) Two most used are 'TH Niramit AS' and '[TH Sarabun New](http://www.f0nt.com/release/th-sarabun-new/)'.

volzhs commented:
@Noshyaar This uses Waree font for Thai.
![image](https://user-images.githubusercontent.com/8281454/29826442-440a3116-8d12-11e7-87f9-aeb7748ebe7b.png)
it looks wrong baseline as before this PR.

Noshyaar commented:
I prefer the proportion of Waree font. Although some might argue that Noto has better readability because it emphasizes on the heads. Waree should look better on HiDPI as that's not the case.

Wish there's a way to fix the baseline consistently. From that image, -2 y offset is all it needs. Font size is okay.

(You can use the word "อรุณสวัสดิ์" to test ascender and descender.)

akien-mga commented:
@volzhs Can you make the corresponding PR for master?

volzhs commented:
@akien-mga no problem.
